lib/server/monalisaSynchronizers/DataPassesSynchronizer.js

```diff
@@ -0,0 +1,181 @@
+     * Combine logic of fetching data from service like MonALISA,
+     * processing and inserting to database
+     * @returns {Promise<void>} promise
+     */
+    async sync() {
+        const { default: fetch } = await import('node-fetch');
+        const context = await this.getContext();
+
+        return await fetch(this.baseUrl, this.httpOptions)
+            .then((response) => response.json())
+            .then((rawData) => this.processRawResponse(rawData, context))
+            .then((data) => this.executeDbAction(data));
+    }
+
+    /**
+     * Get synchronization context
+     * Fetch lastRun information telling whether some data pass must be updated
+     * @return {object} context
+     */
+    async getContext() {
+        // eslint-disable-next-line function-paren-newline
+        const [lastRunsData] = await sequelize.query(
+            'SELECT name, max(dpr.run_number) as last_run_in_details \
+            FROM data_passes AS dp \
+            LEFT JOIN data_passes_runs AS dpr \
+                ON dpr.data_pass_id = dp.id \
+            GROUP BY name;');
+
+        const lastRunsEntries = lastRunsData.map((lastRunData) => {
+            const { name, last_run_in_details: lastRun } = lastRunData;
+            return [name, lastRun];
+        });
+
+        return {
+            lastRuns: Object.fromEntries(lastRunsEntries),
+            yearLowerLimit: this.yearLowerLimit,
+        };
+    }
+
+    /**
+     * Check wether data of one data pass are valid
+     * @param {object} dataPass data pass data
+     * @param {object} context context of synchronization
+     * @return {boolean} true if data pass is valid
+     */
+    needToUpdateDataPass(dataPass, context) {
+        return dataPass.lhcPeriod?.year >= context.yearLowerLimit &&
+            (!dataPass.lastRun ||
+            dataPass.lastRun !== context.lastRuns[dataPass.name]);
+    }
+
+    /**
+     * Adjust data pass data to desired format
+     * @param {object} dataPass data pass
+     * @return {object} adjusted data pass
+     */
+    adjustSingleDataPass(dataPass) {
+        const { lhcPeriod } = dataPass;
+        dataPass = mapObjectProperties(dataPass, this.fieldsMapping);
+        dataPass.outputSize = dataPass.outputSize ? Number(dataPass.outputSize) : null;
+        dataPass.lhcPeriod = lhcPeriod;
+        return dataPass;
+    }
+
+    /**
+     * Used to process raw response to custom format, remove unnecessary data
+     * @param {object} rawResponse raw data acquired from external service
+     * @param {object} context context of synchronization
+     * @return {object[]} adjusted data
+     */
+    async processRawResponse(rawResponse, context) {
+        return Object.entries(rawResponse)
+            .map(([dataPassName, dataPass]) => {
+                dataPass['name'] = dataPassName.trim();
+                return dataPass;
+            })
+            .filter((dataPass) => dataPass.name.match(RegExp(`${LHC_PERIOD_NAME_REGEX.source}_`)))
```

Comment:
I would rather use the `extractLhcPeriod` instead of using regex (that should stay module internals).
You can also create a function `isValidLhcPeriodName`

However, it's weird to test if a variable always named datapass is actually a data pass

Reply:
it's not testing for `LHC Period name` but for `Data Pass name` with is different, when `LHC Period name` is always prefix of `Data Pass name` followed by '_'